src/services/itemMembership/ws/hooks.ts

```diff
@@ -1,90 +1,74 @@
+import { ItemMembership } from '../entities/ItemMembership';
+import ItemMembershipService from '../service';
+import { ItemMembershipEvent, itemMembershipsTopic } from './events';
 
-// /**
-//  * Registers real-time websocket events for the item memberships service
-//  * @param websockets Websocket service instance of the server
-//  * @param runner TaskRunner executor of the server
-//  * @param itemService Item database layer
-//  * @param itemMembershipService ItemMembership database layer
-//  * @param itemMembershipTaskManager ItemMembership task manager
-//  * @param validationDbHandler Database transaction handler used to validate subscriptions
-//  */
-// export function registerItemMembershipWsHooks(
-//   websockets: WebSocketService,
-//   runner: TaskRunner<Actor>,
-//   validationDbHandler: DatabaseTransactionHandler,
-// ): void {
-//   websockets.register(itemMembershipsTopic, async (req) => {
-//     const { channel: itemId, member, reject } = req;
-//     // item must exist
-//     const item = await itemService.get(itemId, validationDbHandler);
-//     if (!item) {
-//       reject(NotFound());
-//     }
-//     // member must have at least read access to item
-//     const allowed = await itemMembershipService.canRead(member.id, item, validationDbHandler);
-//     if (!allowed) {
-//       reject(AccessDenied());
-//     }
-//   });
+export function registerItemMembershipWsHooks(
+  repositories: Repositories,
+  websockets: WebsocketService,
+  itemService: ItemService,
+  itemMembershipService: ItemMembershipService,
+): void {
+  websockets.register(itemMembershipsTopic, async (req) => {
+    const { channel: itemId, member } = req;
+    // item must exist with read permission, else exception is thrown
+    await itemService.get(member, repositories, itemId, PermissionLevel.Read);
+  });
 
-//   // on create:
-//   // - notify member of new shared item IF creator != member
-//   // - notify item itself of new membership
-//   const createItemMembershipTaskName = itemMembershipTaskManager.getCreateTaskName();
-//   runner.setTaskPostHookHandler<ItemMembership>(
-//     createItemMembershipTaskName,
-//     async (membership, actor, { handler }) => {
-//       const itemId = getChildFromPath(membership.itemPath);
-//       const item = await itemService.get(itemId, handler);
-//       if (!item) {
-//         return;
-//       }
-//       if (membership.memberId !== item.creator) {
-//         websockets.publish(memberItemsTopic, membership.memberId, SharedItemsEvent('create', item));
-//       }
-//       websockets.publish(itemMembershipsTopic, item.id, ItemMembershipEvent('create', membership));
-//     },
-//   );
+  // on create:
+  // - notify member of new shared item IF creator != member
+  // - notify item itself of new membership
+  itemMembershipService.hooks.setPostHook(
+    'create',
+    async (member, repositories, membership: ItemMembership) => {
```

Comment:
If you type the HookManager instance, you will get the typings here for free (and at any usage site), as well as event names